### DISCUSSION TITLE
Custom Script: variable assignment from another variable

### DISCUSSION THREAD

**Discussion by tux-mania** [netbox-community/netbox#9620] [Q&A] [answered]
Hi all.

I'm creating a Custom Script where I get values given by the user (with StringVar or ObjectVar).
Instead of displaying too much fields for the user, I would like to assign a variable value depending on the result of a variable value given by the user.
For example: from the SiteGroup value, I want to get the Tenant (I made groups in a way I can identify the tenant with the groups' names).

My code is like this:
```
class CreateSite(Script):

    class Meta:
        name = 'Add new site'

    group = ObjectVar(
        model=SiteGroup
        )

    def run(self, data, commit): 

        if data['group'] in ('group1', 'group2'):
            tenant_code = 'tenant1'
        else:
            tenant_code = 'tenant2'

        site = Site(
            name=data['site_name'],
            slug=slugify(data['site_name']),
            group=data['group'],
            status=data['status'],
            tenant=tenant_code
        )

        site.save()
```

When launching the script, I get the following error:
```
ValueError: Cannot assign "'tenant2'": "Site.tenant" must be a "Tenant" instance
```

Even if the error message seems self explanatory, I do not find the error. 'tenant2' is one of the tenants.
Any idea?
Thanks.

**Comment by candlerb** [comment, marked as answer]
In this code:

```
    group = ObjectVar(
        model=SiteGroup
        )
...
        if data['group'] in ('group1', 'group2'):
            tenant_code = 'tenant1'
        else:
            tenant_code = 'tenant2'
```

In the `run()` method, `data['group']` will be an object instance of type SiteGroup, not a string.

Something like this might work:

```
if data['group'].name in ['group1', 'group2']:
```

The opposite problem occurs here:

```
        site = Site(
            name=data['site_name'],
            slug=slugify(data['site_name']),
            group=data['group'],
            status=data['status'],
            tenant=tenant_code   <<<
        )
```

You can't assign a string to the 'tenant' attribute.  It has to be an instance of a Tenant object.  Alternatively, I think you can set `tenant_id`, but then that has to be an integer (primary key).

**Reply by tux-mania** [reply]
Thank you so much!!!
Everything works as expected with your solution. 
Thanks again for the solution and the explanation!
Cheers!
